5-WebApp-AuthZ/5-2-Groups/README-incremental-instructions.md

```diff
@@ -0,0 +1,274 @@
+
+> Did the sample not work for you as expected? Did you encounter issues trying this sample? Then please reach out to us using the [GitHub Issues](../../../../issues) page.
+
+### Processing Groups claim in tokens, including handling **overage**
+
+#### The `groups` claim
+
+The object id of the security groups the signed in user is member of is returned in the `groups` claim of the token.
+
+```JSON
+{
+  ...
+  "groups": [
+    "0bbe91cc-b69e-414d-85a6-a043d6752215",
+    "48931dac-3736-45e7-83e8-015e6dfd6f7c",]
+  ...
+}
+```
+
+#### Groups overage claim
+
+To ensure that the token size doesn’t exceed HTTP header size limits, the Microsoft Identity Platform limits the number of object Ids that it includes in the **groups** claim.
+
+If a user is member of more groups than the overage limit (**150 for SAML tokens, 200 for JWT tokens**), then the Microsoft Identity Platform does not emit the groups claim in the token. Instead, it includes an **overage** claim in the token that indicates to the application to query the [Graph API](https://graph.microsoft.com) to retrieve the user’s group membership.
+
+```JSON
+{
+  ...
+  "_claim_names": {
+    "groups": "src1"
+    },
+    {
+   "_claim_sources": {
+    "src1": {
+        "endpoint":"[Graph Url to get this user's group membership from]"
+        }
+    }
+  ...
+}
+```
+
+> An Identity Office Hours session covered Azure AD App roles and security groups, featuring this scenario and how to handle the overage claim. Watch the video [Using Security Groups and Application Roles in your apps](https://www.youtube.com/watch?v=V8VUPixLSiM)
+
+> You can use the `BulkCreateGroups.ps1` provided in the [App Creation Scripts](./AppCreationScripts/) folder to create a large number of groups and assign users to them. This will help test overage scenarios during development.
+
+##### Order of processing the overage claim
+
+1. Check for the claim `_claim_names` with one of the values being `groups`. This indicates overage.
+
+1. If found, make a call to the endpoint specified in `_claim_sources` to fetch user’s groups.
+
+1. If none found, look into the `groups`  claim for user’s groups.
+
+> When attending to overage scenarios, which requires a call to [Microsoft Graph](https://graph.microsoft.com) to read the signed-in user's group memberships, you app will need to have the [Directory.Read.All](https://docs.microsoft.com/en-us/graph/permissions-reference#group-permissions) for the [getMemberObjects](https://docs.microsoft.com/en-us/graph/api/user-getmemberobjects?view=graph-rest-1.0) function to execute successfully.
```

Comment:
Typo: "to read the signed-in user's group memberships, **you** app will need to have". Should be your app...